### ISSUE TITLE
Bug in OpenbookQA Search Example?

### ISSUE THREAD

**Issue by richardzhuang0412** [willccbb/verifiers#27]
Hi! Thanks for the great work!

I followed the installation guide and run:
`accelerate launch --config_file /path/to/deepspeed_zero3.yaml --num_processes 7 verifiers/examples/openbookqa_search.py`
without any change to the codebase

The training is able to start and process, but I got the following logs:
<img width="1198" alt="Image" src="https://github.com/user-attachments/assets/24d0a19b-e6ad-452a-905b-eb28a4ea12a2" />

Two questions:
1. How come the loss becomes negative?
2. I saw the loss changing though, why do I get `UserWarning: None of the inputs have requires_grad=True. Gradients will be None`?

**Comment by richardzhuang0412** [comment]
Another minor thing: Seems like I've been getting DuckDuckGo rate limit quite often. Does anyone know what the actual rate limit is? Was trying to do some search and fail to find relevant information.

**Comment by willccbb** [comment]
1. Loss in GRPO isn't interpreted the same as in SFT, it always starts at 0, and depends both on advantage estimates and the KL regularization penalty. Rewards are the important metric to watch, not Loss.

2. Yeah, the duckduckgo rate limit can sometimes be an issue depending on batch size. There isn't a great unlimited + free web search API that I'm aware of, unfortunately. I would view this example not as literally what you should expect to train with (and get great results), but rather an illustration of how to use the library with search tools. You'd likely want to replace the search function with any function of your own which calls an API using a key that you have set. This will probably need to be a paid API key in order to have high enough limits for doing RL, the [Brave](https://brave.com/search/api/) API is maybe a good one to check out.

**Comment by richardzhuang0412** [comment]
Really appreciate the timely response! Definitely has resolved all of my issues.
